src/runtime/nitro/plugins/00-context.ts

```diff
@@ -0,0 +1,24 @@
+import { type HeaderMapper, getHeaderValueFromOptions, SECURITY_HEADER_NAMES } from "../../utils/headers"
+import { createRouter} from "radix3"
+import { defineNitroPlugin } from '#imports'
+
+export default defineNitroPlugin((nitroApp) => {
+    const router = createRouter()
+
+    nitroApp.hooks.hook('nuxt-security:headers', (route, headersConfig) => {
```

Comment:
I have a feeling this would be more useful if it was called on each request, otherwise the user should just use route rules?

Reply:
Do you mean it would be up to us to call a hook insteading of hooking into the event ? 

I also thought of this. The reason I made nuxt-security hooking into it instead of calling the hook is because we would need to `await` all hooks. 

So if users are doing a long async hooks, this would block their response but it also makes sense !

Reply:
I need to dig into the code a bit deeper to answer this, think I'm missing some context. Will get back to you when I have a chance (otherwise feel free to ping me)

Reply:
:+1: feel free to ping/call me, maybe can we try to find a better implementation together with @dargmuesli and @Baroshem